vortex-array/src/array/chunked/mod.rs

```diff
@@ -97,7 +97,13 @@ impl ChunkedArray {
                 panic!("Search sorted failed in find_chunk_idx: {}", err);
             });
         let index_chunk = match search_result {
-            SearchResult::Found(i) => i,
+            SearchResult::Found(i) => {
+                if i == self.nchunks() {
+                    i - 1
+                } else {
```

Comment:
feels like there are a lot of bugs of this flavor

Reply:
Yes, I thought this was wrong in couple of places but it was hidden by multiple other bugs. Need to add a new api